CMake/windows_config.cmake

```diff
@@ -40,6 +40,39 @@ macro(os_set_flags)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
         set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} /MP")
 
+        ###############
+        # According to SDLE we need to add the following flags for additional security:
+        # Debug & Release:
+        # /Gy: Enables function-level linking to reduce executable size.
+        # /DYNAMICBASE: Enables Address Space Layout Randomization (ASLR) to improve security.
+        # /GS: Enables buffer security checks to prevent buffer overflows.
+
+        # Release only:
+        # /WX: Treats all warnings as errors.
+        # /LTCG (/GL): Enables Link Time Code Generation to improve performance.
+        # /sdl: Enables additional security checks.
+        # /NXCOMPAT: Enables Data Execution Prevention (DEP) to prevent code execution in data areas.	
+		
+        # see https://readthedocs.intel.com/SecureCodingStandards/2023.Q2.0/compiler/c-cpp/ for more details
+
+        set(ADDITIONAL_COMPILER_FLAGS "/Gy /DYNAMICBASE /GS /wd4101")
+		
+        if(CMAKE_BUILD_TYPE STREQUAL "Debug")
+            message(STATUS "Configuring for Debug build")
+        else() # Release, RelWithDebInfo, or multi configuration generator is being used (aka not specifing build type, or building with VS)
+            message(STATUS "Configuring for Release build")
+            set(ADDITIONAL_COMPILER_FLAGS "${ADDITIONAL_COMPILER_FLAGS} /WX /sdl") 
+            set(CMAKE_LINKER_FLAGS "${CMAKE_LINKER_FLAGS} /INCREMENTAL:NO /LTCG /NXCOMPAT") # ignoring '/INCREMENTAL' due to '/LTCG' specification
+        endif()
+		
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${ADDITIONAL_COMPILER_FLAGS}")
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${ADDITIONAL_COMPILER_FLAGS}")
+		
+        set_directory_properties(PROPERTIES DIRECTORY third-party/ COMPILE_OPTIONS "/W0")
```

Comment:
Let's double check if this is needed and if not, we can lower the warning to a level that does work and is higher than 0, same for Unix/Win

Reply:
Moved the third party exclusion to its own cmake, and ignore rsutils instead